cis194/week2/tien/homework2/src/LogAnalysis.hs

```diff
@@ -0,0 +1,26 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import           Log
+import           Text.Read (readMaybe)
+
+parseMessage :: String -> LogMessage
+parseMessage logStr =
```

Comment:
As a cheery on top, we can have a look at view pattern extension:
https://gitlab.haskell.org/ghc/ghc/wikis/view-patterns
, which will make this more elegant. But don't worry too much, If you can't make it work.